### PR TITLE
Feature post aggregations

### PR DESCRIPTION
**Pull request by tannerlinsley**
.post()
.sortByKey()
.limit()
.squash()
.change()
.changeMap()

### CODE REVIEW

File: src/postAggregation.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+module.exports = function(service) {
+  return {
+    post: post,
+    sortByKey: sortByKey,
+    limit: limit,
+    squash: squash,
+  }
+
+  function post(query, parent, cb) {
+    query.data = parent.locked ? _.clone(parent.data) : parent.data
```
Comment (etimberg) [inline src/postAggregation.js]: Feel like we should split this common line out. It's used everywhere and we always want it to be the same. Maybe a `getData` method?

Comment (etimberg) [inline src/postAggregation.js]: Are queries automatically locked when they are used by more than 1 child?

**Comment by etimberg** [comment]
Looks pretty good. Does this support string queries or only objects?

**Comment by tannerlinsley** [comment]
I have no idea why all of those tests just failed

**Comment by tannerlinsley** [comment]
@etimberg I think it can only accept objects, since they have to map exactly to the `value` object in each result

File: src/postAggregation.js (hunk 1 above)
Comment (tannerlinsley) [inline src/postAggregation.js]: Yeah, we should split that out.  And no, but they should be.  I'll build that in.

**Comment by etimberg** [comment]
I'm ok to merge once tests pass

**Pull request merged**